File: src/copilot_console/app/routers/help.py
"""POST /api/help/ask — answer questions about Copilot Console using the bundled docs."""

from __future__ import annotations

from fastapi import APIRouter, HTTPException
from pydantic import BaseModel, Field

from copilot_console.app.services import help_service

router = APIRouter(prefix="/help", tags=["help"])


class HelpRequest(BaseModel):
    question: str = Field(..., min_length=1, max_length=4000)


class HelpResponse(BaseModel):
    answer: str
    session_id: str


@router.post("/ask", response_model=HelpResponse)
async def ask(request: HelpRequest) -> HelpResponse:
    try:
        result = await help_service.ask_help(request.question)
        return HelpResponse(**result)
    except ValueError as e:
        raise HTTPException(status_code=400, detail=str(e))
    except RuntimeError as e:
        raise HTTPException(status_code=502, detail=str(e))

File: frontend/src/api/help.ts
export interface HelpResponse {
  answer: string;
  session_id: string;
}

const API_BASE = '/api';

export async function askHelp(question: string): Promise<HelpResponse> {
  const response = await fetch(`${API_BASE}/help/ask`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ question }),
  });
  if (!response.ok) {
    let detail = `Help request failed (${response.status})`;
    try {
      const data = await response.json();
      if (data?.detail) detail = String(data.detail);
    } catch {
      // ignore
    }
    throw new Error(detail);
  }
  return response.json();
}
